File: backend/components/metiers/services_personne/services/samples/sample_service.js
// sample_service.js – Service ultra avancé pour ServicesPersonne (JS)
// Expose endpoints, logique métier, validation, audit, etc.

function helloSample(req, res) {
  res.json({ message: "Hello from ServicesPersonne sample service!" });
}

function validateSample(req, res) {
  if (!req.body || !req.body.value) {
    return res.status(400).json({ error: "Missing value" });
  }
  res.json({ validated: true, input: req.body });
}

module.exports = {
  helloSample,
  validateSample
};

File: backend/components/metiers/ressources_humaines/services/samples/sample_service.py
"""
Service Python d'exemple ultra avancé pour Ressources_humaines, endpoints, logique métier, validation, audit, etc.
"""

from fastapi import APIRouter, HTTPException

router = APIRouter()


@router.get("/samples/hello")
def hello_sample():
    """Endpoint de test ultra avancé."""
    return {"message": "Hello from Ressources_humaines sample service!"}


@router.post("/samples/validate")
def validate_sample(data: dict):
    """Validation avancée d'un échantillon."""
    if not data.get("value"):
        raise HTTPException(status_code=400, detail="Missing value")
    return {"validated": True, "input": data}
